modules/private_data_fusion_network/README.md

```diff
@@ -0,0 +1,95 @@
+# Data Fusion Private Network Setup
+
+This module can be used to deploy the networking pre-reqs for a 
+[Private IP Data Fusion Instance](https://cloud.google.com/data-fusion/docs/how-to/create-private-ip).
+Namely:
+- VPC Network
+- Custom subnet to deploy Dataproc clusters in
+- IP allocation for the Peering with the data fusion tenant project VPC
+- VPC Peering to tenant project
+- SSH ingress rule to allow the Data Fusion Instance to reach Dataproc clusters on port 22.
+
+## Usage
+
+Basic usage of this module is as follows:
+
+```hcl
+module "private_data_fusion_network" {
+  source  = "terraform-google-modules/data-fusion/google//modules/private_data_fusion_network"
+  version = "~> 0.1"
+
+  project_id  = "your-gcp-project-id"
+  vpc_network = "data-fusion-net"
+}
+```
+
+Functional examples are included in the
+[examples](../../examples/) directory.
+
+<!-- BEGINNING OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
+## Inputs
+
+| Name | Description | Type | Default | Required |
+|------|-------------|:----:|:-----:|:-----:|
+| data\_fusion\_service\_account | The Google managed Data Fusion Service account | string | n/a | yes |
+| dataproc\_cidr | CIDR range for the subnetwork for Dataproc clusters controlled by private Data Fusion instance | string | `"10.2.0.0/16"` | no |
+| dataproc\_subnet | Name of the subnetwork for Dataproc clusters controlled by private Data Fusion instance | string | `"dataproc-subnet"` | no |
+| instance | Private Data Fusion instance ID | string | n/a | yes |
+| project\_id | GCP Project ID to house resources for private Data Fusion setup | string | n/a | yes |
+| region | GCP region to create subnetwork for peering with private Data Fusion instance | string | `"us-central1"` | no |
+| tenant\_project | Private Data Fusion instance ID | string | n/a | yes |
+| vpc\_network | VPC network to be created or configured with firewall rules and peering for private Data Fusion instance | string | n/a | yes |
```

Comment:
Since this creates a new network, should we call it `vpc_network_name` and make description clear?

Reply:
Does it have to be a name or can it be a self link? If it can be either, the common name is simply "network" in most CFT modules and TF resources.

Reply:
+1

I would just call it network_name